chain/chain/src/resharding.rs

```diff
@@ -235,61 +264,38 @@ mod tests {
                 },
             );
             trie_update.commit(StateChangeCause::Resharding);
-            let (_, trie_changes, _) = trie_update.finalize().unwrap();
+            let (_, trie_changes, state_changes) = trie_update.finalize().unwrap();
             let mut store_update = tries.store_update();
-            let state_root =
-                tries.apply_all(&trie_changes, ShardUId::single_shard(), &mut store_update);
+            let state_root = tries.apply_all(&trie_changes, shard_uid, &mut store_update);
+            let flat_state_changes = FlatStateChanges::from_state_changes(&state_changes);
+            flat_state_changes.apply_to_flat_state(&mut store_update, shard_uid);
             store_update.commit().unwrap();
             state_root
         };
 
-        let num_shards = 4;
-        let account_id_to_shard_id = &|account_id: &AccountId| ShardUId {
-            shard_id: (hash(account_id.as_ref().as_bytes()).0[0] as NumShards % num_shards) as u32,
-            version: 1,
-        };
-
         // add accounts and receipts to the split shards
-        let mut split_state_roots = {
-            let trie_items = tries
-                .get_view_trie_for_shard(ShardUId::single_shard(), state_root)
-                .get_trie_items_for_part(PartId::new(0, 1))
-                .unwrap();
-            let split_state_roots: HashMap<_, _> = (0..num_shards)
-                .map(|shard_id| {
-                    (ShardUId { version: 1, shard_id: shard_id as u32 }, Trie::EMPTY_ROOT)
-                })
-                .collect();
-            let (store_update, split_state_roots) = tries
-                .add_values_to_split_states(
```

Comment:
it would still be valuable to add some values to the trie for testing

Reply:
This part of the test function `let mut split_state_roots = {...}` is exactly the old resharding logic. I've instead abstracted it out to use `Chain::build_state_for_split_shards()` function instead.

Accounts are added in the previous `for account_id in account_ids.iter() {...}` block above.